File: typo-tester-kotlin/src/kotlin/adventureworks/DomainInsertImpl.kt
package adventureworks

import adventureworks.public.*
import typo.runtime.internal.RandomHelper
import java.util.Random

class DomainInsertImpl : TestDomainInsert {
    override fun publicAccountNumber(random: Random): AccountNumber {
        return AccountNumber(RandomHelper.alphanumeric(random, 10))
    }

    override fun publicFlag(random: Random): Flag {
        return Flag(random.nextBoolean())
    }

    override fun publicMydomain(random: Random): Mydomain {
        return Mydomain(RandomHelper.alphanumeric(random, 10))
    }

    override fun publicName(random: Random): Name {
        return Name(RandomHelper.alphanumeric(random, 10))
    }

    override fun publicNameStyle(random: Random): NameStyle {
        return NameStyle(random.nextBoolean())
    }

    override fun publicPhone(random: Random): Phone {
        return Phone(RandomHelper.alphanumeric(random, 10))
    }

    override fun publicShortText(random: Random): ShortText {
        return ShortText(RandomHelper.alphanumeric(random, 10))
    }

    override fun publicOrderNumber(random: Random): OrderNumber {
        return OrderNumber(RandomHelper.alphanumeric(random, 10))
    }
}

File: typo-tester-typo-scala-old/src/scala/adventureworks/DomainInsertImpl.scala
package adventureworks

import adventureworks.public.*
import typo.runtime.internal.RandomHelper

import java.util.Random

class DomainInsertImpl extends TestDomainInsert {
  override def publicAccountNumber(random: Random): AccountNumber =
    AccountNumber(RandomHelper.alphanumeric(random, 10))

  override def publicFlag(random: Random): Flag =
    Flag(random.nextBoolean())

  override def publicMydomain(random: Random): Mydomain =
    Mydomain(RandomHelper.alphanumeric(random, 10))

  override def publicName(random: Random): Name =
    Name(RandomHelper.alphanumeric(random, 10))

  override def publicNameStyle(random: Random): NameStyle =
    NameStyle(random.nextBoolean())

  override def publicPhone(random: Random): Phone =
    Phone(RandomHelper.alphanumeric(random, 10))

  override def publicShortText(random: Random): ShortText =
    ShortText(RandomHelper.alphanumeric(random, 10))

  override def publicOrderNumber(random: Random): OrderNumber =
    OrderNumber(RandomHelper.alphanumeric(random, 10))
}
